neuralpp/test/quick_tests/symbolic/expression_test.py

```diff
@@ -137,6 +137,21 @@ def test_basic_function_application():
         func3, [constant_two, BasicFunctionApplication(func2, [constant_two, constant_two])]))
 
 
+def test_basic_quantifier_expressions():
+    i = BasicVariable('i', int)
+    i_range = (0 < i) & (i < 10)
+    sum_ = BasicSummation(int, i, i_range, i)
+    assert sum_.subexpressions[0].syntactic_eq(BasicConstant(operator.add, Callable[[int, int], int]))
+    assert sum_.subexpressions[1].syntactic_eq(i)
+    assert sum_.subexpressions[2].syntactic_eq(i_range)
+    assert sum_.subexpressions[3].syntactic_eq(i)
+    assert len(sum_.subexpressions) == 4
+    a = BasicVariable('a', int)
+    new_sum = sum_.replace(i, a)
+    assert sum_.syntactic_eq(BasicSummation(int, i, (0 < i) & (i < 10), i))
+    assert new_sum.syntactic_eq(BasicSummation(int, a, (0 < a) & (a < 10), a))
+
+
```

Comment:
Can we have a test here for the `set` method as well? I'd also love to see an example of nested `QuantifierExpression`, which is a very common use case.

Reply:
Yes!